libresapi/src/api/FileSharingHandler.h

```diff
@@ -22,14 +22,21 @@
 #include "StateTokenServer.h"
 
 #include <retroshare/rsfiles.h>
+#include <retroshare/rsnotify.h>
+#include <util/rsthreads.h>
 
 namespace resource_api
 {
 
-class FileSharingHandler: public ResourceRouter
+class FileSharingHandler: public ResourceRouter, NotifyClient
 {
 public:
-	FileSharingHandler(StateTokenServer* sts, RsFiles* files);
+	FileSharingHandler(StateTokenServer* sts, RsFiles* files, RsNotify* notify);
+	~FileSharingHandler();
+
+	// from NotifyClient
```

Comment:
Please use Doxigen style comments for headers documentation like https://github.com/RetroShare/RetroShare/blob/master/libretroshare/src/gxstrans/p3gxstrans.h#L210